packages/apps/intervention/src/api/equipment-api.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2024 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  createStandardSearch,
+  getSearchCriterias,
+} from '@axelor/aos-mobile-core';
+import {Equipment} from '../types';
+
+const createEquipmentsCriteria = ({
+  searchValue,
+  inService,
+  partnerId,
+  isPlaceEquipment = false,
+  parentPlaceId,
+}) => {
+  const criteria = [getSearchCriterias('intervention_equipment', searchValue)];
+
+  criteria.push({
+    fieldName: 'partner.id',
+    operator: '=',
+    value: partnerId,
+  });
+
+  if (inService != null) {
+    if (inService) {
+      criteria.push({
+        fieldName: 'inService',
+        operator: '=',
+        value: inService,
+      });
+    } else {
+      criteria.push({
+        fieldName: 'inService',
+        operator: 'isNull',
+      });
+    }
```

Comment:
```
      criteria.push({
        fieldName: 'inService',
        operator: inService ? '=' : '!=',
        value: true,
      });
```

I think it will manage false and null with the same criteria this way. inService is initialized as null but might become false if user click two times on the checkbox